### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by CelaniDe**
I have created a Dockerfile at the project root directory and have included detailed instructions in the accompanying README file. The decision to create Dockerfile stemmed from challenges encountered during the local development installation process.

### CODE REVIEW

File: Docker [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+FROM node:14
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the package.json and package-lock.json files to the container
+COPY package*.json ./
+
+# Install project dependencies
+RUN npm install --global @gridsome/cli
```
Comment (dpliakos) [inline Docker:11]: Please reorder `RUN npm install --global @gridsome/cli` to be before `COPY package*.json ./`. The latter is expected to require execution more often than the gridsome CLI installation

File: Docker [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+# Copy the rest of your application code to the container
+COPY . .
+
+RUN npm install
+
+# Expose the port that Gridsome uses (default is 8080)
```
Comment (dpliakos) [inline Docker:19]: Please update the comment to accurately describe the function of the expose command.

"[...] The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container [...]
> References https://docs.docker.com/engine/reference/builder/#expose

**Review by dpliakos — CHANGES_REQUESTED** [review]
Well done @CelaniDe ! Thanks for your contribution! 

I'm glad you managed to overcome the development setup issues using docker, but I would highly recommend not using it as a development tool for this app. The sb-website repository has a single service and all of its dependencies can be managed by npm. 

Given that context, I believe that docker adds complexity to the development environment rather than simplifying it. New contributors and students with no prior experience with containerization may be intimidated by the usage of docker. 

I suggest using `nvm` for solving the issues with the current setup until we upgrade the repo's dependencies [1]

In case you decide to keep the docker-based development setup, please add a docker-compose configuration. It uses a path-based context, making container management way simpler than plain docker CLI. Also, a `.dockerignore` file would help reduce the image build time

Please let me know if you need any further help with the setup or changes

> References
[1] I made a pull request that sets up nvm. Please try and verify that that setup works https://github.com/ieee-ihuthess/sb-website/pull/20

File: Docker (hunk 2 above)
Comment (CelaniDe) [reply Docker:19]: Thank you for your response @dpliakos . I concur with your assessment that Docker may not be the most practical choice for development purposes. However, I suggested the utilization of Docker primarily due to its ability to offer a seamless and consistent environment across various machines, alleviating concerns related to npm or other technology dependencies. In my view, incorporating Docker, or alternatively, a Docker Compose file within our repository aligns with established best practices observed in many GitHub repositories.

**Review by dpliakos — APPROVED** [review]
Thanks for providing the updates!

**Pull request merged**